src/screens/WalletScreen/WalletScreen.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useMemo } from 'react';
+import { AssetList } from '../../components/asset-list';
+import { Page } from '../../components/layout';
+import { navbarHeight } from '@/components/navbar/Navbar';
+import { Box } from '@/design-system';
+import { useAccountAccentColor, useAccountSettings, useWalletSectionsData } from '@/hooks';
+import { Toast, ToastPositionContainer } from '@/components/toasts';
+import { useRecoilValue } from 'recoil';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { addressCopiedToastAtom } from '@/recoil/addressCopiedToastAtom';
+import { RemoteCardsSync } from '@/state/sync/RemoteCardsSync';
+import { RemotePromoSheetSync } from '@/state/sync/RemotePromoSheetSync';
+import { MobileWalletProtocolListener } from '@/components/MobileWalletProtocolListener';
+import Routes from '@/navigation/Routes';
+import { useWalletCohort } from '@/hooks/useWalletCohort';
+import { useRemoveScreen } from '@/hooks/useRemoveFirstScreen';
+import { useInitializeWalletAndSetParams } from '@/hooks/useInitiailizeWalletAndSetParams';
+import { useLoadDeferredData } from '@/hooks/usLoadDeferredWalletData';
+import { useAppIconIdentify } from '@/hooks/useIdentifyAppIcon';
+
+const UtilityComponents = React.memo(() => (
+  <>
+    <RemoteCardsSync />
+    <RemotePromoSheetSync />
+    <MobileWalletProtocolListener />
+  </>
+));
+
+function WalletScreen() {
+  const { network: currentNetwork, accountAddress } = useAccountSettings();
+  const insets = useSafeAreaInsets();
+
+  const {
+    isWalletEthZero,
+    isLoadingUserAssets,
+    isLoadingBalance,
+    briefSectionsData: walletBriefSectionsData,
+  } = useWalletSectionsData({ type: 'wallet' });
+
+  useWalletCohort();
+  useRemoveScreen(Routes.WELCOME_SCREEN);
+  useInitializeWalletAndSetParams();
+  useLoadDeferredData();
+  useAppIconIdentify();
+
+  const isAddressCopiedToastActive = useRecoilValue(addressCopiedToastAtom);
+  const isLoadingUserAssetsAndAddress = isLoadingUserAssets && !!accountAddress;
+  const { highContrastAccentColor } = useAccountAccentColor();
+
+  const disableRefreshControl = useMemo(
+    () => isLoadingUserAssetsAndAddress || isLoadingBalance,
+    [isLoadingUserAssetsAndAddress, isLoadingBalance]
+  );
+
+  return (
+    <Box as={Page} flex={1} testID="wallet-screen">
+      <Box style={{ flex: 1, marginTop: -(navbarHeight + insets.top) }}>
+        <AssetList
+          accentColor={highContrastAccentColor}
+          disableRefreshControl={disableRefreshControl}
+          isWalletEthZero={isWalletEthZero}
+          network={currentNetwork}
+          walletBriefSectionsData={walletBriefSectionsData}
+        />
+      </Box>
+      <ToastPositionContainer>
+        <Toast isVisible={isAddressCopiedToastActive} text="􀁣 Address Copied" testID="address-copied-toast" />
+      </ToastPositionContainer>
```

Comment:
Ideally this is in its own component to prevent top level re-render when address is copied.